### PR TITLE
Set up server skeleton and dependencies.

### PR DESCRIPTION
**Pull request by jinnyz**
Created files for basic server and added useful dependencies.

### CODE REVIEW

File: server/src/app.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+var path = require('path');
+var cookieParser = require('cookie-parser');
+var logger = require('morgan');
+
+var indexRouter = require('./routes/index');
+var usersRouter = require('./routes/users');
+
+var app = express();
+const port = 3000;
+app.listen(port, () => console.log("Listening on port 3000"));
```
Comment (yaoius) [inline server/src/app.js:12]: usually this is the last thing in the file, after setting all your middleware

File: server/src/app.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// error handler
+app.use(function(err, req, res, next) {
+  // set locals, only providing error in development
+  res.locals.message = err.message;
+  res.locals.error = req.app.get('env') === 'development' ? err : {};
+
+  res.status(err.status || 500).send();
+});
+
+module.exports = app;
```
Comment (yaoius) [inline server/src/app.js:37]: don't need to export this given this is our top level

File: server/src/app.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+app.listen(port, () => console.log("Listening on port 3000"));
+
+app.use(logger('dev'));
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
+
+app.use('/', indexRouter);
+app.use('/users', usersRouter);
```
Comment (yaoius) [inline server/src/app.js:21]: are we keeping these templates here? it's fine if you want to but they'll just need to be removed later

File: server/src/app.js (hunk 2 above)
Comment (jinnyz) [reply server/src/app.js:37]: removed, think this was also leftover from the generation script

File: server/src/app.js (hunk 1 above)
Comment (jinnyz) [reply server/src/app.js:12]: Moved lines 11 + 12 to the bottom

File: server/src/app.js (hunk 3 above)
Comment (jinnyz) [reply server/src/app.js:21]: I can modify these later for routers we'll actually use later. But not removing these now.

**Pull request merged**